### PR TITLE
Spain autonomous communities and provinces

### PR DESCRIPTION
**Pull request by Rukant**
I just changed the maps.json file and added the new areas to the corresponding folder. Check if everything is fine specially the nominatim query because I don't know if those are the correct params.

### CODE REVIEW

**Review by BilelJegham — CHANGES_REQUESTED** [review]
Thanks for your contribution;
Please follow the README (https://github.com/GeoGuess/GeoGuess-Maps/blob/main/README.md)

You should edit `src/list.maps.json` or `src/list.areas.json`, not maps.json ;)

**Comment by Rukant** [comment]
Ok, the maps.json file is back to it's original state and the changes are now in the src/list.areas.json. For the step of the generation of the i18n keys I don't know if I need to do something else.

**Comment by Rukant** [comment]
Syntax error in the JSON file is fixed.

File: public/geojson/areas/spain_autonomous_communities.geojson [hunk 1]
```diff
@@ -0,0 +1 @@
...25029,42.60862811280103],[-2.8199145754944723,42.614559569814155],[-2.8221313444975795,42.58894645998477],[-2.8406044195234763,42.58166694455957],[-2.8376487275193334,42.5711520889454],[-2.8173283449908464,42.55767150482467],[-2.798116346963914,42.57384820576954],[-2.780382194939053,42.57951005110025],[-2.771884580427141,42.59622597540996],[-2.7737318879297312,42.60701044270654],[-2.7637564274157462,42.62291753196901],[-2.730504892369133,42.617525298320714],[-2.7061204333349487,42.59973092728135],[-2.6806275897992116,42.594608305315475],[-2.6861695123069804,42.58301500297164],[-2.6913419733142323,42.54284286229186],[-2.705381510333913,42.516690529097644],[-2.6854305893059447,42.520195480969036],[-2.67730243629455,42.5102198487197],[-2.656612592265546,42.50401878002416],[-2.6499622852562243,42.48541557393755],[-2.6285335182261846,42.49566081786931],[-2.592326291175427,42.4918862543155],[-2.5779172926552274,42.482989068795824],[-2.537276527598255,42.48325868047824],[-2.538015450599291,42.49404314777482],[-2.519542375573394,42.508062955260385],[-2.514739376066661,42.4899989725386],[-2.4770543030138317,42.48676363234963],[-2.4489752289744704,42.497008876281384],[-2.4386303069599684,42.
```
Comment (BilelJegham) [inline public/geojson/areas/spain_autonomous_communities.geojson:1]: Unfortunately, we have an issue in this geojson, some "provincia" values don't match with OpenStreetMap :

1. Received from openstreetmap "Navarra" but "Comunidad Foral de Navarra"  in geojson
Tested with https://nominatim.openstreetmap.org/reverse?format=json&lat=42.63008462628674&lon=-1.6433664720849774&zoom=5&addressdetails=1&accept-language=es
2. Received from openstreetmap "Asturias" but "Principado de Asturias" in geojson
Tested with https://nominatim.openstreetmap.org/reverse?format=json&lat=43.27224333834266&lon=-5.946760248088729&zoom=5&addressdetails=1&accept-language=es

File: src/list.areas.json [hunk 2]
```diff
@@ -198,5 +198,37 @@
...
+            "pathKey": "NAME_1",
+            "nominatimResultPath": "address.state",
+            "nominatimQueryParams": {
+                "zoom": "5",
+                "addressdetails": "1",
+                "accept-language": "es"
+            }
+        }
+    },
+    "spain-provinces": {
```
Comment (BilelJegham) [inline src/list.areas.json:218]: Unfortunately, we have an issue in this geojson, some "provincia" values don't match with OpenStreetMap :
1. Received from openstreetmap "Valencia" but "València"  in geojson
Tested with https://nominatim.openstreetmap.org/reverse?format=json&lat=39.4271294470365&lon=-0.8663281562502531&zoom=8&addressdetails=1&accept-language=es
2. Received from openstreetmap "Gerona" but "Girona" in geojson
Tested with https://nominatim.openstreetmap.org/reverse?format=json&lat=42.124818507961336&lon=2.729617274154627&zoom=8&addressdetails=1&accept-language=es
2. Received from openstreetmap "Alicante" but "Alacant" in geojson
Tested with https://nominatim.openstreetmap.org/reverse?format=json&lat=38.52134644552799&lon=-0.37103329879899116&zoom=8&addressdetails=1&accept-language=es
2. Received from openstreetmap "Región de Murci" but "Murcia" in geojson
Tested with https://nominatim.openstreetmap.org/reverse?format=json&lat=37.82459302432988&lon=-1.3477247855292498&zoom=8&addressdetails=1&accept-language=es
2. Received from openstreetmap "Lérida" but "Lleida" in geojson
Tested with https://nominatim.openstreetmap.org/reverse?format=json&lat=42.06838324616863&lon=1.1442427212390878&zoom=8&addressdetails=1&accept-language=es

**Review by BilelJegham — CHANGES_REQUESTED** [review]
Hi, thx for the fix. 
Tested both GeoJSON, we have smalls issues with labels.

Best regards

**Comment by Rukant** [comment]
I've changed it to polygon in type because probably OpenStreetMap will have a lot of problems with that geojson file. I guess creating the areas with OpenStreetMap have some advantages, so I can try to change to do the changes to the geojson file in the future. Apart from the discrepancy in regional languages and spanish, some regions don't have a state_district field when it's the same as the state one. Let me know if that is a problem and, in that case, if there is a way we can bypass it.

**Review by BilelJegham — APPROVED** [review]

**Comment by BilelJegham** [comment]
Ok, 
OpenStreetMap is more precise than polygon for borders. But polygon works fine.

Thx for your contribution. Congrats for your 1st pull request :confetti_ball:

**Pull request merged**